Gems/AWSGameLift/Scripts/gamelift_build_package_linux.py

```diff
@@ -0,0 +1,169 @@
+                print("Package cancelled.")
+                return
+            elif (val != 'Y'):
+                print("Package cancelled. Incorrect user response.")
+                return
+            else:  # delete everything and start fresh
+               shutil.rmtree(package_dir)
+               os.mkdir(package_dir)
+    else:
+        os.mkdir(package_dir)
+    
+    # this script takes a minute to run, display an animation
+    done = False
+    animated_str = "loading"
+    def animate():
+        for c in itertools.cycle(['|', '/', '-', '\\']):
+            if done:
+                break
+            # clear the line and then display animated text
+            sys.stdout.write("\033[K")
+            print(f'{animated_str} {c}', end="\r")
+            time.sleep(0.1)
+        sys.stdout.write("\033[K")
+        print('Done!', end="\r")
+
+    t = threading.Thread(target=animate)
+    t.start()
+
+    # make required sub-folders
+    animated_str = "make new folders"
+    os.mkdir(os.path.join(package_dir, "assets"))
+    os.mkdir(os.path.join(package_dir, "lib64"))
+
+    # copy executables
+    for filename in os.listdir(build_bin_folder):
+        # skip gamelauncher, editor libraries, and assetprocessor
+        if any(map(filename.__contains__, {'.Editor.so', '.GameLauncher', 'libEditor'})):
+            continue
+        if filename in {'AssetProcessor', 'Editor', 'MaterialEditor', 'o3de', 'libComponentEntityEditorPlugin.so'}:
+            continue
+
+        f = os.path.join(build_bin_folder, filename)
+        if os.path.isfile(f) and os.access(f, os.X_OK):
+            animated_str = f"copying: {filename}"
+            target = os.path.join(package_dir, filename)
+            shutil.copyfile(f, target)
+            # preserve read/write/execute permission. this is lost during the copy.
+            st = os.stat(f)
+            os.chmod(target, st.st_mode | stat.S_IEXEC)
+
+    # copy registry folder
+    animated_str = "copying: registry"
+    target = os.path.join(package_dir, "Registry")
+    shutil.copytree(os.path.join(build_bin_folder, "Registry"), target)
+
+    # copy asset cache folder
+    animated_str = f"copying: asset cache"
+    target = os.path.join(package_dir, "assets", "linux")
+    shutil.copytree(os.path.join(project_cache_folder, "linux"), target)
+
+    # copy libraries from /usr/lib/x86_64-linux-gnu/ to <package base folder>/lib64/
+    regex = re.compile('(libc\+\+.*)|(libxkb.*)|(libxcb.*)|(libX.*)|(libbsd.*)')
+    for file in os.listdir("/usr/lib/x86_64-linux-gnu/"):
+        if regex.match(file):
+            animated_str = f"copying: {file}"
+            target = os.path.join(package_dir, 'lib64', file)
+            shutil.copyfile(os.path.join('/usr/lib/x86_64-linux-gnu/', file), target)
+
+    # create a build install script for GameLift
+    with open(os.open(os.path.join(package_dir, "install.sh"), os.O_CREAT | os.O_WRONLY, 0o777), "w") as f:
+        f.write("""#!/bin/sh 
```

Comment:
Nit: consider moving script to al2_install.sh and then piping in.  Consider adding comments to shell script.